packages/manager/apps/pci-load-balancer/src/pages/delete/Delete.page.tsx

```diff
@@ -0,0 +1,94 @@
+import { Translation, useTranslation } from 'react-i18next';
+import { DeletionModal } from '@ovh-ux/manager-pci-common';
+import { useNavigate, useParams } from 'react-router-dom';
+import { OsdsText } from '@ovhcloud/ods-components/react';
+import { ODS_TEXT_LEVEL, ODS_TEXT_SIZE } from '@ovhcloud/ods-components';
+import { ODS_THEME_COLOR_INTENT } from '@ovhcloud/ods-common-theming';
+import { useNotifications } from '@ovh-ux/manager-react-components';
+import { useContext } from 'react';
+import { ShellContext } from '@ovh-ux/manager-react-shell-client';
+import { ApiError } from '@ovh-ux/manager-core-api';
+import {
+  useAllLoadBalancers,
+  useDeleteLoadBalancer,
+} from '@/api/hooks/useLoadBalancer';
+
+export default function DeletePage() {
+  const { addSuccess, addError } = useNotifications();
+  const { t: tDelete } = useTranslation('delete');
+  const { t } = useTranslation('');
+  const navigate = useNavigate();
+  const { projectId, loadBalancerId } = useParams();
+  const {
+    data: loadBalancers,
+    isPending: isPendingLoadBalancers,
+  } = useAllLoadBalancers(projectId);
+  const onClose = () => {
+    navigate('..');
+  };
+  const loadBalancer = loadBalancers.find((lb) => lb.id === loadBalancerId);
+
+  const {
+    deleteLoadBalancer,
+    isPending: isPendingDelete,
+  } = useDeleteLoadBalancer({
+    projectId,
+    loadBalancer,
+    onError(error: ApiError) {
+      addError(
+        <Translation ns="common">
+          {(_t) =>
+            _t('octavia_load_balancer_global_error', {
+              message: error?.response?.data?.message || error?.message || null,
+              requestId: error?.config?.headers['X-OVH-MANAGER-REQUEST-ID'],
+            })
+          }
+        </Translation>,
+        true,
+      );
+      onClose();
+    },
+    onSuccess() {
+      addSuccess(
+        <Translation ns="common">
+          {(_t) => _t('octavia_load_balancer_delete_success')}
+        </Translation>,
+        true,
+      );
+      navigate('..');
+    },
+  });
+
+  const onConfirm = () => {
+    deleteLoadBalancer();
+  };
+
+  const onCancel = () => {
+    navigate('..');
+  };
+
+  const isPending = isPendingLoadBalancers || isPendingDelete;
```

Comment:
not agree  the fetch of the whole list and then find your LB target before rendering the modal, at least fetch the needed LB by it's id